p2p/proto/snapshot.proto

```diff
@@ -0,0 +1,109 @@
+syntax = "proto3";
+
+import "p2p/proto/common.proto";
+import "p2p/proto/state.proto";
+
+message PatriciaNode {
+    message Edge {
+        uint32  length = 1;
+        Felt252 path   = 2;  // as bits of left/right
+        Felt252 value  = 3;
+    }
+    message Binary {
+        Felt252 left  = 1;
+        Felt252 right = 2;
+    }
+
+    oneof node {
```

Comment:
I'm not sure how to interpret this. Can you add some comment on how is this expected to work?

Reply:
It is using the same structure as here: https://github.com/starkware-libs/cairo-lang/blob/master/src/starkware/cairo/common/patricia.cairo. Essentially a binary node is when there are two sub paths (keys) that differ by a bit (so one is left, one is right) and an edge is when there is some path that is unique.